### PR TITLE
[stdlib] Removed SIMD.splat

### PR DESCRIPTION
**Pull request by devesh-2002**
Issue #2942,
I have removed the .splat method from SIMD. Please look into it, and let me know if there are any changes needed.

### CODE REVIEW

**Comment by JoeLoser** [comment]
!sync

File: docs/manual/parameters/index.ipynb [hunk 1]
```diff
@@ -282,7 +283,7 @@
     "can call the `splat()` method like this:\n",
     "\n",
     "```mojo\n",
-    "GenericArray[Float64].splat(8, 0)\n",
+    "GenericArray[Float64](8, 0)\n",
```
Comment (arthurevans) [inline docs/manual/parameters/index.ipynb:286]:
```suggestion
    "GenericArray[Float64].splat(8, 0)\n",
```

Comment (arthurevans) [reply docs/manual/parameters/index.ipynb:286]: We need to revert this line, this is not being called on SIMD; this is calling a static method declared above on LL277-278 demonstrating the use of the `Self` type.

**Comment by JoeLoser** [comment]
!sync

**Pull request closed without merging**